frontend-react/src/components/USLink.tsx

```diff
@@ -1,38 +1,103 @@
-import React, { PropsWithChildren } from "react";
+import React, { AnchorHTMLAttributes } from "react";
 import { Link, NavLink } from "react-router-dom";
+import classnames from "classnames";
 
-// Known issue with the `PropsWithChildren generic in React 18,
-// so I wrote this in a way that we can just remove `<any>` and be
-// okay if we update our React version.
-interface USNavLinkProps extends PropsWithChildren<any> {
-    href: string;
+/** React.PropsWithChildren has known issues with generic extension in React 18,
+ * so rather than using it here, we are using our own definition of child types.
+ * One less headache when updating to React 18 in the future! */
+interface CustomLinkProps {
+    children: React.ReactNode;
+    anchor?: boolean;
     className?: string;
     activeClassName?: string;
 }
-type USLinkProps = Omit<USNavLinkProps, "activeClassName">;
+type USLinkProps = AnchorHTMLAttributes<{}> &
+    Omit<CustomLinkProps, "activeClassName">;
+type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
-/** A single link for rendering standard links */
-export const USLink = ({ children, href, className }: USLinkProps) => (
-    <Link to={href} className={`usa-link ${className}`}>
-        {children}
-    </Link>
-);
+/** A single link for rendering standard links. Uses a `Link` by default
+ * but adding `anchor` will make this a generic anchor tag.
+ * @example
+ * <USLink href="/page">To Page</USLink> // uses <Link> from react-router-dom
+ * <USLink anchor href="#this-section-on-my-page">To Section</USLink> // uses <a>
+ * */
+export const USLink = ({
+    anchor = false,
```

Comment:
I'm wondering if we can infer whether to use an `<a>` versus `<Link>` if 1) the component is rendered within React Router context or 2) the `href` starts with `http(s)://`, `#`, `mailto:`, `tel:` (or whatever other protocol we might want to use) 🤔 

Just a thought, though.  I don't think it hurts to have this override prop!

Reply:
Hmm, great point about the inference! Unless it's hidden better than Waldo himself, we don't use `tel:`. I think it's safe to condense it to just `starts with "http" || "#" || "mailto:"`. That said, in the spirit of no pushes needed for this kinda thing, is there a definitive list of protocols we'd want to make an enumeration of for this detection? Don't see why we shouldn't do it all in one swift commit. :)

Reply:
`href`s can basically take any URI scheme so it's unfortunately a pretty big list 😬 I'd say that the most popular ones that we'd likely use are the ones we've already discussed, though.  Plus a `javascript:` one just in case (even though that's generally not great practice!).

So anything that starts with:
- `http(s)://`
- `mailto:`
- `tel:` (I'm thinking we might as well add it just in case 🤷)
- `javascript:`
- `#`

But again, definitely not a blocking point!  Just some food for thought :)

Reply:
I'll go ahead and ticket/link this one as well as a good follow-up ticket :)